core/metrics-core-service/src/main/java/org/hawkular/metrics/core/service/tags/TagsConverter.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2014-2017 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.core.service.tags;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+import org.hawkular.metrics.model.param.Tags;
+
+/**
+ * The string format is a list of tags in the {@code
+ * name:value} form, comma-separated.
+ *
+ * @author Thomas Segismont
+ * @author Stefan Negrea
+ */
+public class TagsConverter {
```

Comment:
@jotak, I made a mistake in the last comment. The converter in the API is still needed because there is still a considerable number of APIs that still accept Tags as a parameter.